operators/go/submariner_controller.go.nolint

```diff
@@ -99,57 +102,110 @@ func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Re
 	//subm_engine_sa.Name = "submariner-engine"
 	//reqLogger.Info("Created a new SA", "SA.Name", subm_engine_sa.Name)
 
-	// Define a new Pod object
-	// TODO: Make this responsive to size
-	pod := newPodForCR(instance)
+	deployment := newDeploymentForCR(instance)
 
 	// Set Submariner instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, deployment, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+	foundDeployment := &appv1.Deployment{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, foundDeployment)
 	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
+		reqLogger.Info("Creating a new Deployment",
+			"Deployment.Namespace", deployment.Namespace, "Deployment.Name", deployment.Name)
+		err = r.client.Create(context.TODO(), deployment)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 
-		// Pod created successfully - don't requeue
 		return reconcile.Result{}, nil
 	} else if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Pod already exists - don't requeue
-	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	reqLogger.Info("Skip reconcile: Deployment already exists",
+		"Deployment.Namespace", foundDeployment.Namespace, "Deployment.Name", foundDeployment.Name)
 	return reconcile.Result{}, nil
 }
 
+func newDeploymentForCR(cr *submarinerv1alpha1.Submariner) *appv1.Deployment {
+
+	labels := map[string]string{
+		"app":       "submariner-engine",
+		"component": "engine",
+	}
+
+	replicas := int32(1)
```

Comment:
This is where we’d take the size into account, assuming it makes any sense whatsoever. Does the engine even support multiple instances?

Reply:
There should be engine per gateway node (in case you have more than one) but we can do that in the future

Reply:
Right, yes, I translated exactly what helm does, but I believe that's something we can base on size. Then it will pick N replicas between the labeled nodes.

Reply:
I have added a comment on that to the commit message.